src/Lib/BSP/IO/BspFileReader.cs

```diff
@@ -0,0 +1,210 @@
+            {
+                Lump lump = l.Item1;
+                LumpHeader header = l.Item2;
+                if (lump is GameLump)
+                {
+                    gameLump = lump;
+                    gameLumpHeader = header;
+                }
+                else if (gameLump is not null && header.Offset != 0 && header.Offset != gameLumpHeader.Offset)
+                {
+                    gameLumpHeader.UncompressedLength = header.Offset - gameLumpHeader.Offset;
+                    Console.WriteLine($"changed gamelump length to {gameLumpHeader.Length}");
+                    break;
+                }
+            }
+        }
+
+        //sort by offset so the output file looks more like the input
+        private void SortLumps()
+        {
+            Dictionary<BspLumpType, Lump<BspLumpType>> newLumps = new();
+            foreach (var l in Lumps.OrderBy(x => x.Item2.Offset))
+            {
+                var temp = _bsp.Lumps.Where(x => x.Value == l.Item1).First();
+                newLumps.Add(temp.Key, temp.Value);
+                _bsp.Lumps.Remove(temp.Key);
+            }
+            if (_bsp.Lumps.Any())
+                throw new InvalidDataException("bsp lumps and reader headers didn't match!");
+            _bsp.Lumps = newLumps;
+
+        }
+
+        //for testing
+        private bool CheckOverlapping()
+        {
+            bool ret = false;
+            Lump<BspLumpType> prevLump = null;
+            LumpHeader prevHeader = null;
+            bool first = true;
+            foreach (var l in Lumps.OrderBy(x => x.Item2.Offset))
+            {
+                var lump = (Lump<BspLumpType>)l.Item1;
+                LumpHeader header = l.Item2;
+                if (first)
+                {
+                    first = false;
+                    prevLump = lump;
+                    prevHeader = header;
+                }
+                else if (header.Length > 0)
+                {
+                    long prevEnd = prevHeader.Offset + prevHeader.Length;
+                    if (header.Offset < prevEnd)
+                    {
+                        Console.WriteLine($"lumps {prevLump.Type} and {lump.Type} overlapping");
+                        if (prevLump.Type == BspLumpType.Game_lump)
+                            Console.WriteLine("but the previous lump was GAME_LUMP and the length is a lie");
+                        else
+                            ret = true;
+                    }
+                    else if (header.Offset > prevEnd)
+                    {
+                        Console.WriteLine($"space between lumps {prevLump.Type} {prevEnd} <-- {header.Offset - prevEnd} --> {header.Offset} {lump.Type}");
+                    }
+                    if (header.Offset + header.Length >= prevEnd)
+                    {
+                        prevLump = lump;
+                        prevHeader = header;
+                    }
```

Comment:
I disagree. The condition affects those 2 lines. A continue affects the entire loop. Sure it would work here but if anyone comes back to this code an wants to add something later they would need think about why the loop should stop here and than probably change it to a if again. Also the only reason you don't have to indent is because you hide the continue in the same line as the if. I don't think this change makes it easier to read and I don't like one line if statements unless its as simple as "if(skip) continue;"

Reply:
Okay, fair enough, I suppose doesn't actually read better the way I was suggesting